packages/dbos-compiler/cli.ts

```diff
@@ -137,7 +137,7 @@ program
       const compileResult = compile(tsconfigPath, options.suppressWarnings);
       if (compileResult) {
         const [dbosConfig] = parseConfigFile(options);
-        await deployToDatabase(dbosConfig.poolConfig, compileResult, options.appVersion);
+        await deployToDatabase(dbosConfig.poolConfig!, compileResult, options.appVersion);
```

Comment:
seems like we should verify poolConfig is present rather than asserting it like this.

Reply:
`parseConfigFile` always fills in the poolConfig object

Reply:
Then why is it declared as optional in the return type?

Reply:
You mean in the signature of parseConfigFile?

`function parseConfigFile(cliOptions?: ParseOptions): [DBOSConfig, DBOSRuntimeConfig]`

or in the declaration of the `DBOSConfig` interface?

I made all fields optional in the later, for backward compatibility

Reply:
handling in #807